homeassistant/components/openweathermap/services.py

```diff
@@ -0,0 +1,35 @@
+"""Services for OpenWeatherMap."""
+
+from __future__ import annotations
+
+from homeassistant.core import HomeAssistant, ServiceCall, SupportsResponse
+
+from . import WeatherUpdateCoordinator
+from .const import ATTR_API_MINUTELY_FORECAST, DOMAIN, OWM_MODE_V30
+
+SERVICE_GET_MINUTELY_FORECAST = f"get_{ATTR_API_MINUTELY_FORECAST}"
+
+
+async def async_setup_services(
+    hass: HomeAssistant,
+    mode: str,
+    weather_coordinator: WeatherUpdateCoordinator,
+) -> None:
+    """Set up OpenWeatherMap services."""
+
+    def handle_get_minutely_forecasts(call: ServiceCall) -> None:
+        """Handle the service action call."""
+        return weather_coordinator.data[ATTR_API_MINUTELY_FORECAST]
+
+    if mode == OWM_MODE_V30:
+        hass.services.async_register(
+            domain=DOMAIN,
+            service=SERVICE_GET_MINUTELY_FORECAST,
+            service_func=handle_get_minutely_forecasts,
+            supports_response=SupportsResponse.ONLY,
+        )
+    else:
+        hass.services.async_remove(
+            domain=DOMAIN,
+            service=SERVICE_GET_MINUTELY_FORECAST,
+        )
```

Comment:
Services should be registered in `async_setup` and should not be removed on runtime

Reply:
This is only called when the user makes a change to the integration configuration (`Settings` -> `Devices & Services` -> `OpenWeatherMap` -> `Configure`) and changes the forecast mode of the integration to a mode which does not support Minute forecasting. As such, this isn't really on runtime: it's only when the user reconfigures the integration without first deleting it.

If the service isn't removed, the user will be returned an empty forecast - with no feedback as to the reason - when using an unsupported mode. I would argue that is worse UX.

Reply:
If the mode of the user is incorrect, you can raise a ServiceValidationError, explaining why it failed. More info about this can be found for now in https://github.com/home-assistant/developers.home-assistant/pull/2400/files